### PR TITLE
REST routes updates and Https support for validation

### PR DESCRIPTION
**Pull request by MikhailRoot**
+Webroot parsing fix
- in admin community link validation now works with https too.
- REST route for getting community's categories list used in admin side in comments settings updated to API v1 to work.

### CODE REVIEW

**Comment by linc** [comment]
You haven't synced to current master, so you are double-adding the webroot fix and it's conflicting. Also, see my comments on the other PR. And please be sure to give a descriptive title, not "Fixes".

File: widgets.php [hunk 1]
```diff
@@ -47,9 +47,9 @@ function vf_widget_discussions($args) {
 			
 		$url = vf_get_value('url', $options, '');
 		$link_url = vf_get_link_url($options);
-		$resturl = array($url, '?p=discussions.json');
+		$resturl = array($url, '/discussions.json');
 		if ($categoryid > 0)
-			$resturl = array($url, '?p=categories/'.$categoryid.'.json');
+			$resturl = array($url, '/categories/'.$categoryid.'/');
```
Comment (linc) [inline widgets.php]: Did you mean to remove the `.json` here?

**Comment by MikhailRoot** [comment]
ok i need to make a new pull request with routes done in more universal way without /api/ ?

File: widgets.php (hunk 1 above)
Comment (MikhailRoot) [inline widgets.php]: sorry  i think i accidentally  deleted it. i'll recheck all changes.

**Comment by MikhailRoot** [comment]
I've checked all changes again, and all seems ok. Also updated Pull request  title to make it more informative.

**Comment by MikhailRoot** [comment]
Sorry i'm a bit newbie with some of git features. I've merged webroot fix in mine so now as i see it could be merged successfully. 
Also added accidentally removed .json  ending in  route what you've mentioned above.

**Comment by linc** [comment]
Thanks! Good catch on the root problem too, I didn't realize we were still using the `p` param over here.

**Pull request merged**